apiary-metastore-events/kafka-metastore-events/kafka-events-common/src/main/java/com/expediagroup/apiary/extensions/events/metastore/common/io/jackson/JsonMetaStoreEventSerDe.java

```diff
@@ -0,0 +1,133 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expediagroup.apiary.extensions.events.metastore.common.io.jackson;
+
+import static com.fasterxml.jackson.databind.DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+
+import org.apache.hadoop.hive.metastore.api.MetaException;
+import org.apache.hadoop.hive.metastore.api.SkewedInfo;
+import org.apache.hadoop.hive.metastore.events.CreateTableEvent;
+import org.apache.hadoop.hive.metastore.events.ListenerEvent;
+import org.apache.thrift.TBase;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+
+import com.expediagroup.apiary.extensions.events.metastore.common.event.EventType;
+import com.expediagroup.apiary.extensions.events.metastore.common.event.SerializableListenerEvent;
+import com.expediagroup.apiary.extensions.events.metastore.common.io.MetaStoreEventSerDe;
+
+public class JsonMetaStoreEventSerDe implements MetaStoreEventSerDe {
+  private static final Logger log = LoggerFactory.getLogger(JsonMetaStoreEventSerDe.class);
+
+  private final ObjectMapper mapper;
+
+  public JsonMetaStoreEventSerDe() {
+    SimpleModule thriftModule = new SimpleModule("ThriftModule");
+    registerSerializers(thriftModule);
+    registerDeserializers(thriftModule);
+    mapper = new ObjectMapper();
+    mapper.configure(FAIL_ON_UNKNOWN_PROPERTIES, false);
+    mapper.registerModule(thriftModule);
+  }
+
+  private void registerSerializers(SimpleModule module) {
+    module.addSerializer(new SkewedInfoSerializer());
+    module.addSerializer(new JacksonThriftSerializer<>(TBase.class));
+  }
+
+  private void registerDeserializers(SimpleModule module) {
+    module.addDeserializer(SkewedInfo.class, new SkewedInfoDeserializer());
+  }
+
+  @Override
+  public byte[] marshal(SerializableListenerEvent listenerEvent) throws MetaException {
+    try {
+      log.debug("Marshalling event: {}", listenerEvent);
+      ByteArrayOutputStream buffer = new ByteArrayOutputStream();
+      mapper.writer().writeValue(buffer, listenerEvent);
+      byte[] bytes = buffer.toByteArray();
+      if (log.isDebugEnabled()) {
+        log.debug("Marshalled event is: {}", new String(bytes));
+      }
+      return bytes;
+    } catch (IOException e) {
+      String message = "Unable to marshal event " + listenerEvent;
+      log.error(message, e);
```

Comment:
Probably don't need to log this if we're going to throw it as we should be logging the thrown exception later too (let's not repeat ourselves in the log files).